app/models/data.py

```diff
@@ -230,22 +245,28 @@ def lastUpdateEt(self):
 
     @hybrid_property
     def totalTestResults(self):
-        # Calculated value (positive + negative) of total test results.
-        # For consistency with public API, treating a negative null as 0
-
-        # to match the public API, use totalTestEncountersViral for selected states
-        if self.state in ["RI", "CO", "ND"]:
-            return self.totalTestEncountersViral or 0
-
-        # MA uses totalTestsViral
-        if self.state in ["MA"]:
-            return self.totalTestsViral or 0
-
-        if self.negative is None:
-            return self.positive or 0
-        if self.positive is None:
-            return self.negative or 0
-        return self.positive + self.negative
+        """Calculated value of total test results
+
+        This value is determined based on the state's totalTestResultsFieldDbColumn, with empty cells converted to 0.
+        If a CoreData column name is specified, that column will be used for totalTestResults.
+        Alternatively, the 'posNeg' keyword can be used to indicate totalTestResults = (positive+negative)"""
+
+        # there are circumstances where this is called where the state doesn't exist. default to posNeg in those cases
```

Comment:
when would this be called without the state existing? Since we have an FK on the state to all CoreData objects, I think that shouldn't happen...

Reply:
You'd think, but it breaks editing without it. I think that's because we called `to_dict` on CoreData objects that hadn't been flushed to the DB yet, so they didn't have a `state_obj` attached yet. But I wasn't thrilled with this and it's going to cause confusion later, so I fixed the underlying issue instead and took this out, which is far better.